src/power_grid_model/errors.py

```diff
@@ -0,0 +1,62 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+
+"""
+Error classes
+"""
+
+from typing import List, Optional
+
+import numpy as np
+
+from power_grid_model.power_grid_core import power_grid_core as pgc
+
+VALIDATOR_MSG = "\nTry validate_input_data() or validate_batch_data() to validate your data.\n"
+
+
+class PowerGridError(ValueError):
+    pass
+
+
+class PowerGridBatchError(ValueError):
+    failed_scenarios: np.ndarray
+    error_messages: List[str]
+
+
+def find_error() -> Optional[ValueError]:
+    """
+
+    Returns:
+
+    """
+    error_code: int = pgc.err_code()
```

Comment:
I'm not really sure bout this global error handling, can't we supply some handle/instance?

Reply:
The user is not supposed to call this directly. It is part of the back-end core. Since we can't do multi-threading in Python, a global handle is enough.

In the new commit I have moved all core part in a separate `core` folder.

Reply:
What do you mean by we "can't do multi-threading in Python"?